### DISCUSSION TITLE
readd listing public avatar

### DISCUSSION THREAD

**Discussion by ghost** [vrcx-team/VRCX#184] [Q&A] [answered]
i dont know exactly why does listing public avatar feature has been removed from vrcx.it was the very useful feature since most of the avatar creator make their avatar public and they dont know how to make world so atleast from vrcx anyone can easily get public avatar of any user. i hope vrcx dev will going to readd the feature again

**Comment by Myrkie** [comment, marked as answer]
please refer to [issue 182](https://github.com/pypy-vrc/VRCX/issues/182)

its been removed as its now an obsolete feature thanks to an API update 3 hours ago that now makes it impossible to see users public avatars without using "DIRECT ACCESS" Which is still accessible from VRCX

![image](https://user-images.githubusercontent.com/20288698/114511241-a744b700-9c05-11eb-8b69-985384575537.png)

**Reply by ghost** [reply]
but for this feature i need a avatar id and how would i get avatar id from public avatar when the information of specific public avatar has been hidden

**Reply by Myrkie** [reply]
You don't atleast not without a Mod that can copy avatar ids from in game.

**Reply by Myrkie** [reply]
the entire point of this API update by vrchat was to increase API security and make it so random people cannot easily get avatars that were set as public as a lot of people set avatars as public but don't put them on pedestals and keep it as a "semi private" avatar for only select friends.

**Reply by ghost** [reply]
as u said it keep it as a semi private for only friends which added on the list even i couldnt able to browse my friends avatar also

**Reply by Myrkie** [reply]
that was an example, if an avatar is private no one can use it except for the owner when an avatar is semi public in some peoples eyes they want to keep the avatar to themselves and only give it to people they choose, the API update was done to make it so the avatar can only be given out via Cloning the avatar or if the owner shares the ID and you get that ID there is other ways that break vrchat TOS.

**Reply by ghost** [reply]
now what is this when i try to browse my own avatar in vrcx getting this error >>User and userid are mutually exclusive avatars<<404

**Reply by Myrkie** [reply]
> now what is this when i try to browse my own avatar in vrcx getting this error >>User and userid are mutually exclusive avatars<<404

Please refer to [commit 524c8dee79a2e6073988e4298bbeed8abfe32831](https://github.com/pypy-vrc/VRCX/commit/524c8dee79a2e6073988e4298bbeed8abfe32831)

this update removes the unnecessary avatar viewing code and fixes viewing your own avatars

**Reply by ghost** [reply]
i have lot of avatarss but i can view only 10 to 15 avatars as private only and the avatars i made public i cant browse them

**Reply by Myrkie** [reply]
The new update hasn't been released yet your probably viewing from cache

**Reply by ghost** [reply]
ok i see

**Comment by rigal111** [comment]
hello my english not that good, so possible to avoid this "you can browse only your avatar" search avatar function was really good i didnt needed to go world to get avatar i miss this function, is it temporary? or is this fixable?
